src/ess/loki/io.py

```diff
@@ -157,80 +161,51 @@ def merge_background_monitor_runs(
+def load_nexus_sample(file_path: FilePath[Filename[RunType]]) -> RawSample[RunType]:
+    return RawSample[RunType](nexus.load_sample(file_path))
 
-    events = _preprocess_data(
-        dg[f'{data_name}_events'],
-        sample_position=sample_position,
-        source_position=source_position,
+
+def dummy_load_sample(file_path: FilePath[Filename[RunType]]) -> RawSample[RunType]:
+    return RawSample[RunType](
+        sc.DataGroup({'position': sc.vector(value=[0, 0, 0], unit='m')})
     )
-    if data_name in DETECTOR_BANK_RESHAPING:
-        events = DETECTOR_BANK_RESHAPING[data_name](events)
 
-    dg[f'{data_name}_events'] = events
-    return dg
+
+def load_nexus_source(file_path: FilePath[Filename[RunType]]) -> RawSource[RunType]:
+    return RawSource[RunType](nexus.load_source(file_path))
 
 
 def load_nexus_detector(
-    filename: FilePath[Filename[RunType]],
+    file_path: FilePath[Filename[RunType]],
     detector_name: NeXusDetectorName,
-    transform_path: TransformationPath,
-    source_name: NeXusSourceName,
-    sample_name: Optional[NeXusSampleName],
+    raw_source: RawSource[RunType],
+    raw_sample: RawSample[RunType],
 ) -> LoadedSingleFileDetector[RunType]:
-    return LoadedSingleFileDetector[RunType](
-        _load_events(
-            filename=filename,
-            data_name=detector_name,
-            transform_path=transform_path,
-            source_name=source_name,
-            sample_name=sample_name,
-        )
+    out = nexus.load_detector(file_path=file_path, detector_name=detector_name)
+    # Note here we specify the name instead of using
+    # ess.reduce.nexus.extract_detector_data because we need the name to put the
+    # events back into the original data group.
+    key = f'{detector_name}_events'
+    events = _preprocess_data(
+        out[key],
+        sample_position=raw_sample['position'],
+        source_position=raw_source['position'],
     )
+    if detector_name in DETECTOR_BANK_RESHAPING:
+        events = DETECTOR_BANK_RESHAPING[detector_name](events)
+    out[key] = events
+    return LoadedSingleFileDetector[RunType](out)
 
 
 def load_nexus_monitor(
-    filename: FilePath[Filename[RunType]],
+    file_path: FilePath[Filename[RunType]],
     monitor_name: NeXusMonitorName[MonitorType],
-    transform_path: TransformationPath,
-    source_name: NeXusSourceName,
-    sample_name: Optional[NeXusSampleName],
+    raw_source: RawSource[RunType],
 ) -> LoadedSingleFileMonitor[RunType, MonitorType]:
-    return LoadedSingleFileMonitor[RunType, MonitorType](
-        _load_events(
-            filename=filename,
-            data_name=monitor_name,
-            transform_path=transform_path,
-            source_name=source_name,
-            sample_name=sample_name,
-        )
-    )
+    out = nexus.load_monitor(file_path=file_path, monitor_name=monitor_name)
+    key = f'{monitor_name}_events'
+    out[key] = _preprocess_data(out[key], source_position=raw_source['position'])
```

Comment:
Similar question to above, why do we need to preprocess before extracting the monitor data? This makes this brittle, as we still hard-code some naming convention.